1.8.1: Problems with pyfits 3.4

With the new pyfits release, the unit tests show errors like
```
________________________ TestFitsFunc.test_write_map_C _________________________

self = <healpy.test.test_fitsfunc.TestFitsFunc testMethod=test_write_map_C>

    def test_write_map_C(self):
>       write_map(self.filename, self.m, fits_IDL=False)

healpy/test/test_fitsfunc.py:25: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
healpy/fitsfunc.py:153: in write_map
    tbhdu.header.update('PIXTYPE','HEALPIX','HEALPIX pixelisation')
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = XTENSION= 'BINTABLE'           / binary table extension                       ...FORM1  = 'E       '                                                            
args = ('PIXTYPE', 'HEALPIX', 'HEALPIX pixelisation'), kwargs = {}
other = 'PIXTYPE'
update_from_dict = <function update_from_dict at 0x7fa655183ed8>, idx = 0
card = 'P'

    def update(self, *args, **kwargs):
        """[...]"""
    [...]
                else:
                    raise ValueError(
                        'Header update sequence item #%d is invalid; the '
                        'item must either be a 2-tuple containing a keyword '
                        'and value, or a 3-tuple containing a keyword, '
>                       'value, and comment string.' % idx)
E                   ValueError: Header update sequence item #0 is invalid; the item must
            either be a 2-tuple containing a keyword and value, or a 3-tuple containing
            a keyword, value, and comment string.

/usr/lib/python2.7/dist-packages/pyfits/header.py:1017: ValueError
```
According to the pyfits documentation, the calling sequence of pyfits.Header.update() is wrong.
Full error log (also containing the problem described in #328) [here](https://buildd.debian.org/status/fetch.php?pkg=healpy&arch=amd64&ver=1.8.1-1.1%2Bb1&stamp=1456704751)

--- a/healpy/fitsfunc.py
+++ b/healpy/fitsfunc.py
@@ -116,7 +116,7 @@ def write_cl(filename, cl, dtype=np.float64):
 
     tbhdu = pf.new_table(cols)
     # add needed keywords
-    tbhdu.header.update('CREATOR','healpy')
+    tbhdu.header['CREATOR'] = 'healpy'
     writeto(tbhdu, filename)
 
 def write_map(filename,m,nest=False,dtype=np.float32,fits_IDL=True,coord=None,partial=False,column_names=None,column_units=None,extra_header=()):
@@ -223,34 +223,31 @@ def write_map(filename,m,nest=False,dtype=np.float32,fits_IDL=True,coord=None,pa
 
     tbhdu = pf.new_table(cols)
     # add needed keywords
-    tbhdu.header.update('PIXTYPE','HEALPIX','HEALPIX pixelisation')
+    tbhdu.header['PIXTYPE'] = ('HEALPIX', 'HEALPIX pixelisation')
     if nest: ordering = 'NESTED'
     else:    ordering = 'RING'
-    tbhdu.header.update('ORDERING',ordering,
-                        'Pixel ordering scheme, either RING or NESTED')
+    tbhdu.header['ORDERING'] = (ordering,
+                                'Pixel ordering scheme, either RING or NESTED')
     if coord:
-        tbhdu.header.update('COORDSYS',coord,
-                            'Ecliptic, Galactic or Celestial (equatorial)')
-    tbhdu.header.update('EXTNAME','xtension',
-                        'name of this binary table extension')
-    tbhdu.header.update('NSIDE',nside,'Resolution parameter of HEALPIX')
+        tbhdu.header['COORDSYS'] = (coord,
+                                    'Ecliptic, Galactic or Celestial (equatorial)')
+    tbhdu.header['EXTNAME'] = ('xtension',
+                               'name of this binary table extension')
+    tbhdu.header['NSIDE'] = (nside,'Resolution parameter of HEALPIX')
     if not partial:
-        tbhdu.header.update('FIRSTPIX', 0, 'First pixel # (0 based)')
-        tbhdu.header.update('LASTPIX',pixelfunc.nside2npix(nside)-1,
-                            'Last pixel # (0 based)')
-    tbhdu.header.update('INDXSCHM', 'EXPLICIT' if partial else 'IMPLICIT',
-                        'Indexing: IMPLICIT or EXPLICIT')
-    tbhdu.header.update('OBJECT', 'PARTIAL' if partial else 'FULLSKY',
-                        'Sky coverage, either FULLSKY or PARTIAL')
+        tbhdu.header['FIRSTPIX'] = (0, 'First pixel # (0 based)')
+        tbhdu.header['LASTPIX'] = (pixelfunc.nside2npix(nside)-1,
+                                   'Last pixel # (0 based)')
+    tbhdu.header['INDXSCHM'] = ('EXPLICIT' if partial else 'IMPLICIT',
+                                'Indexing: IMPLICIT or EXPLICIT')
+    tbhdu.header['OBJECT'] = ('PARTIAL' if partial else 'FULLSKY',
+                              'Sky coverage, either FULLSKY or PARTIAL')
 
     # FIXME: In modern versions of Pyfits, header.update() understands a
     # header as an argument, and headers can be concatenated with the `+'
     # operator.
     for args in extra_header:
-        if args[0] == 'COMMENT':
-            tbhdu.header.add_comment(*args[1:])
-        else:
-            tbhdu.header.update(*args)
+        tbhdu.header[args[0]] = args[1:]
 
     writeto(tbhdu, filename)
 
@@ -614,7 +611,7 @@ def mwrfits(filename,data,hdu=1,colnames=None,keys=None):
     tbhdu = pf.new_table(cols)
     if type(keys) is dict:
         for k,v in keys.items():
-            tbhdu.header.update(k,v)
+            tbhdu.header[k] = v
     # write the file
     writeto(tbhdu, filename)
 

